### ISSUE TITLE
Long hashes cause path length limit to be exceeded

### ISSUE THREAD

**Issue by nathanaeljones** [conan-io/conan#1154]
> C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppBuild.targets(312,5): error MSB3491: Could not write lines to file "Debug\CompilerIdCXX.tlog\CompilerIdCXX.lastbuildstate". The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters. [C:\Users\BatoPC.conan\data\libjpeg-turbo\1.5.1\imazen\testing\build\9fa351ac1b30ceabd1b27310408605a456b5fd86\libjpeg-turbo-a964424b0195162b7aff6fb951aab5e3953c22d1_build\CMakeFiles\3.8.0-rc3\CompilerIdCXX\CompilerIdCXX.vcxproj]

from https://github.com/imazen/imageflow/issues/126

Is there a way to limit hash length or otherwise reduce this behavior?

**Comment by memsharded** [comment]
There are two ways to deal with long paths in Windows:

- Using the ``short_paths" feature (http://docs.conan.io/en/latest/reference/conanfile/attributes.html#short-paths) in the package recipe with long paths
- From Win10, it is possible to disable the path limits, in the OS. The latest python 3.6 installer already has an option for doing it, but can be done manually.

**Comment by mrexodia** [comment]
For future reference: https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd#enable-long-paths-in-windows-10-version-1607-and-later

```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\FileSystem]
"LongPathsEnabled"=dword:00000001
```

**Comment by memsharded** [comment]
Thanks @mrexodia for that reference, but also a word of caution for users: we have found that even if the OS can support long paths, there are different tools, like build systems and compilers that might still have the same limitations of path length hardcoded. We have been reported a few cases, so the users still had to resort on the Conan short_paths in order to be able to build and use those packages.

**Comment by mrexodia** [comment]
Yeah thanks for the warning! In my case I found this issue while Googling
for an error in another package manager and this fixed it for me 😄
